common/src/main/java/com/microsoft/identity/common/internal/broker/AuthUxJsonPayloadObject.kt

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Data class representing the JSON payload object received from AuthUX.
+ *
+ * @property correlationId The correlation ID for the request.
+ * @property actionName The name of the action being performed.
+ * @property actionComponent The component responsible for the action.
+ * @property params The parameters for the action, including function and data.
+ */
+data class AuthUxJsonPayloadObject(
+    @SerializedName(SerializedNames.CORRELATIONID)
+    val correlationId: String?,
+
+    @SerializedName(SerializedNames.ACTION_NAME)
+    val actionName: String?,
+
+    @SerializedName(SerializedNames.ACTION_COMPONENT)
+    val actionComponent: String?,
+
+    @SerializedName(SerializedNames.PARAMS)
+    val params: AuthUxParams?
```

Comment:
Why is everything nullable? I think one of these fields being optional makes sense but it doesn't make sense for everything to be nullable

Reply:
I think allowing everything to be nullable made sense because we can have specific logging/handling for null fields in the  JSinterface. The object itself is only meant to communicate the payload, so i wanted to leave null handling to the JS interface.

Reply:
But what exactly is the payload contract between client and server? If the contract says that certain fields (for instance actionName) cannot ever be null then we should mark it as such here.

Reply:
Hmm. Not a fan of this custom deserialization. Since you are using class defined in kotlin then why not using kotlin deserialization? I believe that should take care kotlin non-null implicit declarations

Reply:
Discussed offline....

Kotlinx deserialization requires separate dependency and so does Moshi and that can increase size of SDK. We can do it later. For now sticking to GSON. It isn't great but this is what we have in the code today.

Reply:
kotlin deserialization requires adding an additional plugin and dependency. that didn't really seem worth considering we already had Gson and Moshi. For Moshi, there was a similar problem, basically we need another package as a dependency to have moshi working properly for kotlin. Only solution that didn't involve adding an additional dependency was using gson with custom serializer